src/photos/ducks/timeline-clusters/index.jsx

```diff
@@ -95,22 +98,21 @@ const getSectionTitleHours = (dates, index, section, f) => {
 }
 
 /**
- * A section matches if:
- * - The datetime is inside the period.
- * - The section has no period (not a cluster) and is the datetime's day.
+ * A section matches if a not-clustered photo's datetime:
+ * - Is inside its period, or the same day
  */
-const getMatchingSection = (sections, datetime) => {
+const getMatchingSection = (sections, datetime, f) => {
   return Object.keys(sections).find(date => {
     if (sections[date].album) {
-      const startPeriod = new Date(sections[date].album.period.start)
-      const endPeriod = new Date(sections[date].album.period.end)
-      return (
-        (datetime >= startPeriod && datetime <= endPeriod) ||
-        differenceInCalendarDays(datetime, endPeriod) === 0
-      )
+      const startPeriod = sections[date].album.period.start
+      const endPeriod = sections[date].album.period.end
+      const isInsidePeriod =
+        isEqualOrNewer(datetime, startPeriod) &&
+        isEqualOrOlder(datetime, endPeriod)
+      return isInsidePeriod || isSameDay(f, datetime, endPeriod)
```

Comment:
If `datetime` and `endPeriod` are the same day, then doens't that mean `isInsidePeriod ` is true as well?